test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+    ${stdout}    ${rc}=    Execute Command    microshift show-config
+    ...    sudo=True    return_rc=True
+    Log    Microshift command executed. Return Outpu: ${stdout}
+    Should Contain    ${stdout}    ${new_host_ip}
+    Should Be Equal As Integers    ${rc}    0
+
+Check new Kubeconfig generated for new IP
+    [Documentation]    Check the new kubeconfig
+     ${hostname}   ${rc1}=    Execute Command    hostname
+    ...    sudo=True    return_rc=True
+    ${output}    ${rc2}    Execute Command    ls /var/lib/microshift/resources/kubeadmin/${hostname}/
+    ...    sudo=True    return_rc=True
+    Log    Check Microshift new kubeconfig command executed. Return Output: ${output}
+    Should Contain    ${output}    kubeconfig
+    Should Be Equal As Integers    ${rc1}    0
+    Should Be Equal As Integers    ${rc2}    0
+
+Get File Checksum
+    [Arguments]    ${file_path}
+    ${stdout}    ${rc}=    Execute Command     sha256sum ${file_path}
+    ...    sudo=True    return_rc=True
+    Run Keyword If    '${rc}' != '0'    Fail    Failed to get the checksum of the file: ${file_path}
+    [Return]    ${stdout.split()[0]}
+
+Compare File Checksum
+    [Arguments]    ${file_path}    ${expected_checksum}
+    ${current_checksum}=    Get File Checksum    ${file_path}
+    Should Not Be Equal    ${current_checksum}    ${expected_checksum}
+
+*** Test Cases ***
+Verify certificate updates when the host IP changes 
+    [Documentation]  Ensure certificate updates when the host IP changes
+    
+    # Generate new IP in same subnet
+    ${result}=    Generate New IP
+    # Assign the values to global variables
+    Set Global Variable    ${new_host_ip}    ${result}[0]
+    Set Global Variable    ${gateway}       ${result}[1]
+    Set Global Variable    ${device}        ${result}[2]
+    Set Global Variable    ${nmconnection_path}      /etc/NetworkManager/system-connections/${device}.nmconnection
+
+    # Take backup of /etc/hosts and /etc/NetworkManager/system-connections/*.nmconnection
+    ${hosts_backup}=  Save Backup Default File    ${HOSTS_FILE_PATH}
+    Log    Hosts file backed up as: ${hosts_backup}
+
+    ${nmconnection_backup}=  Save Backup Default File    ${nmconnection_path}
+    Log    ens3.nmconnection backed up as: ${nmconnection_backup}
+
+    # Take initial checksum of the external server cert ffile
+    # Save the initial checksum of the file
+    ${initial_checksum}=    Get File Checksum    ${EXTERNAL_SERVER_CERT}
+ 
+    # Add node name and IP to /etc/hosts
+    ${node_name}=    Get NodeName
+    Add Node to Hosts File    ${new_host_ip}    ${node_name}
+
+    # Stop MicroShift
+    Stop MicroShift
+
+    #Modify the NetworkManager configuration using nmcli commands
+    Modify NetworkManager Configuration   ${new_host_ip}/24   ${gateway}   ${device}
+
+    # Reboot the MicroShift system vm
+    Reboot MicroShift System
+    Sleep   200s
```

Comment:
We should not wait here. There is a function in the test library waiting for MicroShift host to be up.

Reply:
https://github.com/openshift/microshift/pull/2151#discussion_r1280278309